optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberDescriptor.java

```diff
@@ -210,7 +230,41 @@ public String getName() {
      * The name does not include generic infomation.
      */
     public String getTypeName() {
-        String typeName = type.getTypeName();
+        String[] holder = new String[1];
+        whenMetadataIsOnField(fd -> {
+            holder[0] = fd.getType();
+        });
+
+        whenMetadataIsOnMethod(md -> {
+            holder[0] = md.getReturnType();
+        });
+
+        if (DescriptorUtils.isPrimitive(holder[0])) {
+            switch (holder[0]) {
+                case "I":
+                    return int.class.getName();
+                case "B":
+                    return byte.class.getName();
+                case "C":
+                    return char.class.getName();
+                case "J":
+                    return long.class.getName();
+                case "F":
+                    return float.class.getName();
+                case "S":
+                    return short.class.getName();
+                case "D":
+                    return double.class.getName();
+                case "Z":
+                    return boolean.class.getName();
+                case "V":
+                    return void.class.getName();
```

Comment:
Doesn't the JDK have anything to translate these for us?

Reply:
Gizmo didn't, but ASM did